routes/v1.js

```diff
@@ -149,37 +228,483 @@ api.declare({
+  });
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, workerType);
+    return err;
+  });
+
+  return p;
+
 });
 
-// List workerTypes
+
+// Delete workerType
+// TODO: send a pulse message that a worker type was removed
+api.declare({
+  method:         'delete',
+  route:          '/worker-type/:workerType',
+  name:           'removeWorkerType',
+  deferAuth:      true,
+  // TODO: Should we have a special scope for workertype removal?
+  scopes:         ['aws-provisioner:get-worker-type:<workerType>'],
+  input:          undefined,  // No input
+  output:         undefined,  // No output
+  title:          "Delete Worker Type",
+  description: [
+    "Delete a WorkerType definition, submits requests to kill all ",
+    "instances and delete the KeyPair from all configured EC2 regions",
+  ].join('\n')
+}, function(req, res) {
+  var ctx         = this;
+  var workerType  = req.params.workerType;
+
+  if(!req.satisfies({
+    workerType:       workerType
+  })) {
+    return; // by default req.satisfies() sends a response on failure, so we're done
+  }
+
+  var p = ctx.WorkerType.load({workerType: workerType})
+
+  p = p.then(function(worker) {
+    return worker.remove();
+  });
+
+  p = p.then(function() {
+    debug('Finished deleting worker type');
+    return res.reply({});
+  });
+
+  p = p.then(function() {
+    //return ctx.publisher.workerTypeDeleted({
+    //  workerType: workerType,
+    //})
+  });
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, workerType);
+    return err;
+  });
+
+  return p;
+});
+
+
 api.declare({
   method:         'get',
   route:          '/list-worker-types',
   name:           'listWorkerTypes',
-  deferAuth:      true,
+  deferAuth:      true, // I don't think we need this unless we do parameterized scopes
   scopes:         [
-    [
-      // Require both scopes... We need to parameterize these for each
-      // <workerType> before we return the result
-      'aws-provisioner:get-worker-type:<workerType>',
       'aws-provisioner:list-worker-types',
-    ]
   ],
   input:          undefined,  // No input
   output:         SCHEMA_PREFIX_CONST + 'list-worker-types-response.json#',
   title:          "List Worker Types",
   description: [
-    // Document what this method does
+    "List all known WorkerType names",
+  ].join('\n')
+}, function(req, res) {
+  var ctx         = this;
+  var workerType  = req.params.workerType;
+
+  if(!req.satisfies({
+    workerType:       workerType
+  })) {
+    return; // by default req.satisfies() sends a response on failure, so we're done
+  }
+
+  var p = this.WorkerType.listWorkerTypes()
+
+  p = p.then(function(workerNames) {
+    return res.reply(workerNames);
+  });
+
+  p = p.catch(function(err) {
+    errorHandler(err, res, 'listing all worker types');
+    return err;
+  });
+
+  return p;
+
+});
+
+
+api.declare({
+  method:         'get',
+  route:          '/get-launch-specs/:workerType',
+  name:           'getLaunchSpecs',
```

Comment:
Well, you could call it that, but it both tests and returns all the launch specs.  since it's both a tester and a getter, I think it's more logical for a getter to be used to test than a tester to get.